IFC-583 improve diff labels

ajtmccarty opened the pull request:
adds more labels to diff responses. there are now labels all over the place
- we've got labels for deleted items
- diff conflicts now include `base_branch_label` and `diff_branch_label`
- diff properties now include `previous_label` and `new_label`
- all these new labels will only be set when appropriate, meaning when the property (or its conflict) is IS_RELATED, HAS_OWNER, or HAS_SOURCE

ajtmccarty committed:
IFC-583 allow getting a deleted node and its attributes

ajtmccarty committed:
a little more unit test

ajtmccarty committed:
IFC-583 include deleted node labels

ajtmccarty committed:
fix unit test

ajtmccarty committed:
add labels all over the place

ajtmccarty committed:
finish adding more labels and testing

dgarros commented on backend/infrahub/core/diff/enricher/labels.py:
```diff
@@ -1,45 +1,198 @@
+        self, display_label_requests: set[DisplayLabelRequest]
+    ) -> dict[str, dict[str, str]]:
+        node_ids = [dlr.node_id for dlr in display_label_requests]
+        query = await NodeGetKindQuery.init(db=self.db, ids=node_ids)
```
Not a blocker for this PR but it would be interesting to understand where these ids come from and see if we could collect the kind at the same time. In most cases, if we are querying the id from a query we should be able to get the kind at the same time.

ajtmccarty replied:
this can be addressed in IFC-584 (add peer kind to relationship element diffs) during the next round of diff updates

dgarros reviewed (commented):
I'm wondering if there's gonna be a bigger implication to making it possible to query deleted node ...
In the previous version of the diff, we were able to query both branches to get the information, have you considered this approach ? is there a drawback to it ?

ajtmccarty committed:
remove invalid conflicts during enrichment

ajtmccarty committed:
include empty diffs in the coordinator to delete them if appropriate

ajtmccarty committed:
fix action determination bug in cardinality one enricher

ajtmccarty committed:
update types for leaf diff values to str | None

ajtmccarty committed:
remove commented out code

ajtmccarty committed:
update handling for cardinality one relationship diffs, add tests

ajtmccarty committed:
mypy fixes

ajtmccarty committed:
Revert "IFC-583 allow getting a deleted node and its attributes"

This reverts commit 32e4bc9e3ee220687e5f6b2d5d0214e106c4e293.

ajtmccarty committed:
formatting

ajtmccarty committed:
update labels without getting deleted nodes

ajtmccarty committed:
fix un-reverted unit test

ajtmccarty commented:
I've reverted the changes to allow getting deleted nodes and updated the DiffLabelsEnricher logic to get the labels from the base or diff branches as appropriated depending on the status of the node: REMOVED nodes will pull from the base branch, ADDED/UPDATED nodes will use the diff branch

ajtmccarty committed:
fix flappy test
